crucible/src/Lang/Crucible/Backend.hs

```diff
@@ -208,6 +209,13 @@ class IsBoolSolver sym where
   -- See also 'addAssertion'.
   addProofObligation :: sym -> Assertion sym -> IO ()
```

Comment:
I think I'd prefer to see a default implementation of `addProofObligation` here in terms of `addDurableProofObligation` rather than have the same default behavior copied into each of the implementations.

Let's also add a comment explaining that "obvious" (concretely true) proof obligations may be discarded.

Reply:
👍